### PR TITLE
Java Update Complete

### PR DESCRIPTION
**Pull request by jacksolirius**
### Change description ###
Upgrading the Adoption repo to JDK 17 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,22 +1,18 @@
 plugins {
-  id 'org.owasp.dependencycheck' version '8.0.2'
+  id 'org.owasp.dependencycheck' version '7.3.0'
```
Comment (jonsoloway-hmcts) [inline build.gradle:2]: This should be set to the latest version, which I believe is currently 8.2.1

File: build.gradle [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
   id 'scala'
-  id 'com.github.lkishalmi.gatling' version '3.0.2'
-  //id 'java-gradle-plugin'
-  id 'groovy'
-  id 'com.gradle.plugin-publish' version '0.9.8'
-  id 'maven-publish'
+  id 'java-library'
+  id 'io.gatling.gradle' version '3.8.4'
```
Comment (jonsoloway-hmcts) [inline build.gradle:5]: Please follow the versions in the table at the top of https://tools.hmcts.net/confluence/display/RQA/Gatling+-+Upgrading+Versions+and+Dependencies as you're using old version of gatling and scala

File: build.gradle [hunk 3]
```diff
@@ -26,51 +22,31 @@ tasks.withType(JavaExec) {
...
+    scala {
+      srcDir 'src/gatling/simulations'
+    }
+    resources {
+      srcDir 'src/gatling/resources'
+    }
+  }
 }
 
 ext ['netty.version'] = '4.0.51.Final'
```
Comment (jonsoloway-hmcts) [inline build.gradle:52]: This isn't required

File: build.gradle [hunk 4]
```diff
@@ -26,51 +22,31 @@ tasks.withType(JavaExec) {
...
-  scalaVersion '2.12.12'
-  simulations = [
-          'simulations.adoptionSimulation'
-  ]
-}
-
-test {
-  testLogging {
-    showStandardStreams = true
+  gatlingVersion '3.8.4'
```
Comment (jonsoloway-hmcts) [inline build.gradle:34]: These are old versions

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.6.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0-bin.zip
```
Comment (jonsoloway-hmcts) [inline gradle/wrapper/gradle-wrapper.properties:3]: This is old, please update

File: gradlew [hunk 6]
```diff
@@ -44,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS='"-Xmx2048m" "-Xms1024m"'
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (jonsoloway-hmcts) [inline gradlew:47]: I'm not sure why you've changed this...

**Review by jonsoloway-hmcts — COMMENTED** [review]
Please see comments.

File: build.gradle (hunk 1 above)
Comment (jacksolirius) [reply build.gradle:2]: Changed to the latest version

File: build.gradle (hunk 2 above)
Comment (jacksolirius) [reply build.gradle:5]: This has been changed

File: build.gradle (hunk 3 above)
Comment (jacksolirius) [reply build.gradle:52]: This has been removed

File: build.gradle (hunk 4 above)
Comment (jacksolirius) [reply build.gradle:34]: Updated to the recent version

File: gradle/wrapper/gradle-wrapper.properties (hunk 5 above)
Comment (jacksolirius) [reply gradle/wrapper/gradle-wrapper.properties:3]: Has been changed to 7.0

File: gradlew (hunk 6 above)
Comment (jacksolirius) [reply gradlew:47]: I've never touched gradlew before so not sure why it changed either, but I have Reverted it back to what it should be

File: src/gatling/simulations/scenarios/adoptionScenario.scala [hunk 7]
```diff
@@ -1581,7 +1581,7 @@ object adoptionScenario {
           .fileName("2MB.pdf")
           .transferEncoding("binary"))
         .asMultipartForm
-        .check(jsonPath("$[0].id").saveAs("Document_ID"))
+        .check(jsonPath("#[0].id").saveAs("Document_ID"))
```
[the remark this answers is not in the document]
Comment (jacksolirius) [reply src/gatling/simulations/scenarios/adoptionScenario.scala:1584]: Has been changed

File: src/gatling/simulations/scenarios/adoptionScenario.scala [hunk 8]
```diff
@@ -1672,10 +1672,10 @@ object adoptionScenario {
 
     .group("AD_945_Card_Details_Check") {
       exec(http("Check Card")
-        .post(PaymentURL + "/check_card/${chargeId}")
+        .post(PaymentURL + "/check_card/#{chargeId}")
         .headers(Headers.paymentHeader)
         .formParam("cardNo", "4444333322221111")
-        .check(jsonPath("$.accepted").is("true")))
+        .check(jsonPath("#.accepted").is("true")))
```
[the remark this answers is not in the document]
Comment (jacksolirius) [reply src/gatling/simulations/scenarios/adoptionScenario.scala:1678]: Has been changed

File: src/gatling/simulations/utils/Environment.scala [hunk 9]
```diff
@@ -2,9 +2,9 @@ package utils
 
 object Environment {
 
-  val baseUrl = "https://adoption-web.${env}.platform.hmcts.net"
-  val idamUrl = "https://idam-web-public.${env}.platform.hmcts.net"
-  val idamAPIURL = "https://idam-api.${env}.platform.hmcts.net"
+  val baseUrl = "https://adoption-web.#{env}.platform.hmcts.net"
+  val idamUrl = "https://idam-web-public.#{env}.platform.hmcts.net"
+  val idamAPIURL = "https://idam-api.#{env}.platform.hmcts.net"
   val paymentUrl = "https://www.payments.service.gov.uk"
```
[the remark this answers is not in the document]
Comment (jacksolirius) [reply src/gatling/simulations/utils/Environment.scala:8]: Has been changed

**Pull request closed without merging**